### ISSUE TITLE
Release 1.5.0

### ISSUE THREAD

**Issue by megies** [obspy/obspy#3228]
This is a very much shortened release check list, after focusing on wheel/conda/pypi only releases.

#### Before release

- [x] Discuss planned release date and update minimum Python and dependency versions accordingly on `master` branch, compare [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html#drop-schedule) and [scipy supported versions](https://docs.scipy.org/doc/scipy/dev/toolchain.html#numpy) (see also [Python release schedule](https://devguide.python.org/versions/)) see #3570 
- [x] [zenodo](https://zenodo.org/): reserve a DOI pre-release (DOI: **10.5281/zenodo.19005357**) **or** change zenodo release workflow
- [x] "What's new?" page in the wiki --> https://github.com/obspy/obspy/wiki/What's-New-in-ObsPy-1.5
- [x] test wheel building by adding test_wheels label to a PR, alternatively test wheel building by tagging a release candidate (source distribution and wheels for release candidates get uploaded to [test.pypi](https://test.pypi.org/project/obspy) if all wheel builds and their test suites succeed)
- [x] make sure Python 3.13 gets added to CI wheel building, should be fixed by #3570 

#### To make the release

- [x] set the [version number and DOI in CHANGELOG.txt](https://github.com/obspy/obspy/blob/master/CHANGELOG.txt), set a git tag, [publish tag as a release](https://github.com/obspy/obspy/tags)
- [x] source distribution to [pypi](https://pypi.org/project/obspy/) ([done automatically by github actions](https://github.com/obspy/obspy/pull/2943))
- [x] wheels for all platforms on [pypi](https://pypi.org/project/obspy/) ([done automatically by github actions](https://github.com/obspy/obspy/pull/2943))
- [x] [conda packages](https://anaconda.org/conda-forge/obspy/files) built via [conda-forge feedstock](https://github.com/conda-forge/obspy-feedstock/pulls)
- [x] [zenodo](https://zenodo.org/): upload sdist zipball and finalize release

#### After release

- [x] Update [docs](https://docs.obspy.org/) once released (should be done automatically, built in gh-actions and [deployed on our server by a bot](https://github.com/obspy/obspy/tree/master/misc/scripts/docs_deploybot)), optionally tar the docs and attach to github release
- [x] Ensure new [docset](https://docsets.obspy.org) is in place (worked automatically through CI ~~currently no docset building set up in release CI~~)
- [x] [News entry in the wiki](https://github.com/obspy/obspy/wiki/#news)
- [x] Update zenodo badge in `README.md` and [wiki sidebar](https://github.com/obspy/obspy/wiki/_Sidebar/_edit)
- [x] Email to the [announcements mailing list](https://mail.python.org/mailman3/lists/obspy.python.org/)
- [x] [Discourse announcement](https://discourse.obspy.org/c/announcements/11)
- [ ] ~~[Tweet](https://twitter.com/obspy) :bird:~~
- [x] [Toot](https://fosstodon.org/@obspy) :elephant: 
- [ ] update base branches for all pull requests bumped to 1.5.x to a new branch `maintenance_1.5.x` (filter those PRs [via github API](https://api.github.com/repos/obspy/obspy/pulls?base=maintenance_1.4.x) or the [pull request search](https://github.com/obspy/obspy/pulls?q=is%3Apr+is%3Aopen+base%3Amaintenance_1.4.x))
- [x] update base branch for dependabot to `maintenance_1.5.x` (in `.github/dependabot.yml`)
- [ ] check dependency versions, some old distros we used to support might have gone EOL since release of 1.4, so we might be able to bump minimum dependency versions of e.g. numpy, scipy, matplotlib, ...: see #3087

---

- [ ] Party :tada: (don't tick before all other checkboxes are ticked!)

**Comment by megies** [comment]
We can start by thinking about what date/month we plan to target, roughly, and accordingly can check if we want to bump minimum Python and dependency versions on master right now already.

**Comment by ThomasLecocq** [comment]
maybe sync with numpy/python release plans ?

**Comment by megies** [comment]
> maybe sync with numpy/python release plans ?

Yep, [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html#drop-schedule) has been what we been looking at for the most part.

**Comment by larsgeb** [comment]
Hey all, just checking in on what the actual status is of the ObsPy project in general. I see the maintenance branches are being updated. How are new releases looking? Totally understand if busy!

**Comment by megies** [comment]
@sannecottaar released a 1.4.2 to fix numpy 2.0 compatibility for until we get the next major release out

**Comment by sannecottaar** [comment]
Cheers!

**Comment by ThomasLecocq** [comment]
I propose we put 1.5.0 on Sep 17 :-) 

> On Sep 16, 2025 drop support for NumPy 1.26 (initially released on Sep 16, 2023) 
[NEP](https://numpy.org/neps/nep-0029-deprecation_policy.html#drop-schedule)

😇

**Comment by ThomasLecocq** [comment]
Hey all,

Here is a dump of my feeling for getting 1.5.0 out: (I fixed 1 Dec 2025 for the release 😇 )

### PRs linked to the Maintenance Branch

Only 4 PRs are open against [maintenance](https://github.com/obspy/obspy/pulls?q=is%3Apr+is%3Aopen+base%3Amaintenance_1.4.x)

I propose to continue like we recently did : review, merge the 3 PRs that are basically ready & then merge into master:
- #3583 
- #3287 
- #2333 - @megies : your call (is it breaking anything?)

### Master Branch
There are 33 Open Issues and PR (85 closed): [list](https://github.com/obspy/obspy/milestone/25)

Many are linked (I did a re-pass on issues & PRs to link them, so we don't forget to close them).

- A few PRs are almost ready, need a little last push & can be merged in to master.
- **All** the stationXML Issues and PRs are completly idle, with NO answer from the original reporter, I propose to move them to a "next release" (bugfix or feature: but I propose to keep them together e.g. a stationXML_fix milestone or else?)
- I moved Mseed3 related issues+PR to milestone 1.6.0 which will be dedicated to that only.

Since master is testing fine for Python 3.8+ and the current "min dep", we can afford to:

- release 1.5.0 with 3.8+ support, and already add deprecation messages for 3.8 AND 3.9 (EOL Today)
- start to build 1.6.0 with 3.10+ support only.

1.5.0 can include 3.8 & 3.9 because #3333 is ready (failing tests are related to the #3628 update): pkg_resources is removed & replaced by importlib.metadata, with a number of small workarounds that will need to be removed immediately after releasing 1.5.0.


@megies @QuLogic @filefolder @barsch @trichter @petrrr and others, please let me know your comments?

**Comment by filefolder** [comment]
thanks for all your work!

I think it's a good idea to retire the maintenance branch ecosystem, and it makes sense to use 1.6 to focus on mseed3. the AR picker I believe was fixed finally in #3552 so we can probably close #2333. there are a few other PRs floating around that would be great to sneak through...

my preference is to try to keep old python compatibility for as long as possible unless there is a serious reason for it, so it would be great to at least keep 1.5 backwards with 3.8.

**Comment by ThomasLecocq** [comment]
As #3333 seems to work, if merged it'll give 1.5 3.8+ compat. Then after that I think it'll be best to go 3.10+, drop the compats and backwards code for numpy <2 etc...

Thanks for checking the other PRs !

**Comment by petrrr** [comment]
Hi all!

I have no strong opinion about dropping older versions of Python. Macports has done  this anyway, so dependencies are gone, all good for py 3.10+ at this point.

On the other hand I would appreciate if we could consider the following for the next release:
* deprecation of  `attach_response` parameter from the fdsn Client;  Now I am not sure issue #2026 is the correct pointer, but the discussion is there.

**Comment by filefolder** [comment]
I agree that `attach_response` needs to be phased out. Easy enough to throw in a depreciation warning for 1.5 and handle it officially on the next one?

**Comment by filefolder** [comment]
seiscomp 7 released today which bumps the xml schema to 0.14. i think i can cram this through by next week.

 #3651

**Comment by claudiodsf** [comment]
👋 **Quick heads-up:** there is now a **working `emscripten-forge` recipe** available for ObsPy:

➡️ https://github.com/emscripten-forge/recipes/tree/main/recipes/recipes_emscripten/obspy

The recipe is maintained by @DerThorsten and @MMesch. 
 
It might be worth considering adding this to the release checklist 😉 .

**Comment by ThomasLecocq** [comment]
Hi all,

We need to release 1.5.0 🚀: 
- the amount of work into the code base is large
- this version maintains python compatibility 3.8+ (which is amazing)
- it could be a short-lived release, with 1.6.0 coming soon after, removing old deprecations and workarounds

The last four PRs that could make it into 1.5.0 are:
- #3580 : I don't work with responses' internals that much, so bit scared of it
- #3653 : I believe it's fine - looks very efficient
- #3622 : same, looks OK
- #3651 : since it's introducing breaking changes 

And if someone could take care of creating a PR for 
- #2026 - it's high time we remove this...

So, calling for a collective effort here, to cross-review & co-work on those last bits.

A note on versions: we'll need to build wheels & conda packages for 3.13 & 3.14 too, so the small changes/edits in #3662 will need to be taken into account in master too.

Pinging @megies @claudiodsf @raciner @filefolder @petrrr @jschaeff @chad-earthscope @d-chambers

**Comment by ThomasLecocq** [comment]
https://github.com/obspy/obspy/wiki/What's-New-in-ObsPy-1.5

**Comment by raciner** [comment]
I would very much like #3589 to be included in one way or the other, as it gives a massive performance boost, at least `_get_response_from_inventory`, as this would probably be the most used part of the code and also the one which would benefit the most. @ThomasLecocq would you appreciate help to get this ready?

**Comment by ThomasLecocq** [comment]
> I would very much like [#3589](https://github.com/obspy/obspy/pull/3589) to be included in one way or the other, as it gives a massive performance boost, at least `_get_response_from_inventory`, as this would probably be the most used part of the code and also the one which would benefit the most. [@ThomasLecocq](https://github.com/ThomasLecocq) would you appreciate help to get this ready?

Hi @raciner thanks! I didn't want to push 'my' PRs into the release, but agree it'd be good to have. Question: as-is it's "working" fine, doesn't change user side... should it go like that and then later be transformed into a sort of _memoization_ of the response call within the Response object, or should it already be changed to that? 


⚠️ For everyone: the idea is to make the rc and release during the week of 2 to 6 March, when I'll be in Munich with @megies

**Comment by ThomasLecocq** [comment]
note I forgot #3633 to the list of "to be part of the" 1.5

**Comment by raciner** [comment]
> Hi [@raciner](https://github.com/raciner) thanks! I didn't want to push 'my' PRs into the release, but agree it'd be good to have. Question: as-is it's "working" fine, doesn't change user side... should it go like that and then later be transformed into a sort of _memoization_ of the response call within the Response object, or should it already be changed to that?
> 
> ⚠️ For everyone: the idea is to make the rc and release during the week of 2 to 6 March, when I'll be in Munich with [@megies](https://github.com/megies)

I don't have a string opinion on that. I would go with whatever is ready by then, provided it is solid. In my opinion, the performance gain of this is worth so much that it would be advisable to have a solid solution for typical work loads right now and instead of waiting for a perfect solution for all cases in two years.

**Comment by megies** [comment]
> I would very much like [#3589](https://github.com/obspy/obspy/pull/3589) to be included in one way or the other

💯 👍

**Comment by megies** [comment]
We actually should briefly discuss minimum dependency and Python versions see #3570.
So far we relatively closely followed [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html#drop-schedule) which is quite aggressive. That would actually mean..

 - dropping Python 3.8, 3.9 and 3.10  (CC #3662) -- if we want be even more aggressive we could even talk about dropping 3.11 which numpy will drop in a few weeks
 - numpy >=2.0 (debatable, maybe should keep 1.26 to have a numpy<2 version supported?)
 - scipy >=1.14 ??
 - matplotlib ~~>= 3.10 ??~~ (EDIT: that would mean only a single matplotlib version which we should avoid)

Any hard feelings about any of these version numbers?

EDIT:

Maybe just go for versions roughly released Autumn 2023 (2.5y before now)?

 - numpy >= 1.26     (initially released Sep 16, 2023)  <- might be problematic for some projects using obspy if we drop numpy 1.x ??
 - scipy >= 1.12     (initially released Jan 20, 2024)
 - matplotlib >= 3.9 (initially released May 16, 2024)

**Comment by ThomasLecocq** [comment]
> Maybe just go for versions roughly released Autumn 2023 (2.5y before now)?
numpy >= 1.26 (initially released Sep 16, 2023) <- might be problematic for some projects using obspy if we drop numpy 1.x ??
scipy >= 1.12 (initially released Jan 20, 2024)
matplotlib >= 3.9 (initially released May 16, 2024)

Agree, py311+ and those, looks reasonable!

**Comment by filefolder** [comment]
we've spent a good deal of effort to keep all those working in 1.5, what is the motivation to rip it all out in the 11th hour?

**Comment by megies** [comment]
It was an oversight that some very old versions were still in master as minimum versions for such a long time, and I'm sorry that that wasn't clear that still some versions would/should get dropped for next release. #3570 was supposed to outline new minimum versions for next release but maybe it was lacking visibility and also by now its already outdated again.

It just makes sense to not support versions of packages that e.g. aren't even supported by their own development teams anymore. Like with Python, 3.8 and 3.9 are already today as we speak out of security support after their full five year life-span, so completely end-of-life. For sure we can keep 3.10 which has half a year security support still if it bothers you to drop it.

All the big packages follow [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html#drop-schedule) in some way or other ([scipy](https://docs.scipy.org/doc/scipy/dev/core-dev/index.html#supported-python-and-numpy-versions), [matplotlib](https://matplotlib.org/devdocs/devel/min_dep_policy.html#dependency-version-policy)), so we should probably also just have it written down somewhere what we actually support, like NEP29 does, for our big three (numpy, scipy, matplotlib) for..
  
 - XX months backwards (e.g. 42 months? which already seems quite long)
 - at least two minor releases

<img height="317" alt="Image" src="https://github.com/user-attachments/assets/4405545c-4310-40ca-9c27-45298c3be803" />

**Comment by megies** [comment]
How does this sound..? "support Python versions that are not end-of-life and minor releases of numpy/scipy/matplotlib that initially released in the last 36 months (3 years)"

 - Python >= 3.10     (initially released Oct 2021)
 - numpy >= 1.25      (initially released Jun 2023)
 - scipy >= 1.11      (initially released Jun 2023)
 - matplotlib >= 3.8  (initially released Sep 2023)

**Comment by megies** [comment]
Right now our CI matrix in #3662 is 55 jobs, which if github ever found out they probably delete our repository without warning 😅

**Comment by claudiodsf** [comment]
Hello,

From my experience, a significant amount of code in the ecosystem still runs on Python 3.8 (and sometimes even 3.7), and would benefit from a bit more continued support from ObsPy. In practice, not all situations are as simple as creating a new Conda or uv environment.

My suggestion would be to keep Python 3.8 support for this release. As @filefolder mentioned, a considerable amount of effort has already gone into maintaining it. Support for 3.8 and 3.9 could then be dropped in the following release if desired.

This approach would reduce risk while also acknowledging the work that has already been invested in maintaining compatibility.

**Comment by ThomasLecocq** [comment]
I have to mention that it's largely my fault that we've been working on this "it works fine 3.8+ let's keep it", and not fixing the minimum versions before starting to push all the PRs , features or bugfixes. Sorry about that, should we have agreed on the bump beforehand, we wouldn't have to do this now.

One important element to keep in mind, and I also overlooked it totally is that when a Release is tagged, it enters an "maintenance" mode (and branch), which HAS to be maintained regularly, for bugfixes etc. With such VERY old versions (unsupported by our dependencies), this adds a lot of risks to have to backport code to our code base, just because we wanted to support more than our dependencies...

**Comment by ThomasLecocq** [comment]
My point is: we are not so many people available for these maintenance efforts, too.

**Comment by ThomasLecocq** [comment]
It's not best practice, nor good practice I guess, but we could allow 3.8-3.10 (no fixing the minimum version in setup.py), but commit to only support bugfixes on 3.11+ in the maintenance branch.

**Comment by claudiodsf** [comment]
Agreed on this last point. My concern is that there are many mission-critical applications for which upgrading the Python version is a significant change.

Moreover, some of these applications might currently run on `master` with Python 3.8 (or 3.9). It would be problematic— to say the least—if version 1.5 suddenly dropped support for those Python versions.

We could instead make this change immediately after the `1.5` tag. Even in maintenance, we could then require Python 3.11+, so that `1.5.1` would already target Python 3.11+.

**Comment by d-chambers** [comment]
Personally, I don't have a strong opinion but I lean more towards dropping unsupported versions of python and dependencies. There is a lot of compatibility code that could be ripped out of ObsPy, leading to a leaner, more robust codebase. 

If ObsPy dropping unsupported python version pushes someone over the edge to upgrade to a faster,  more memory efficient, more secure version of python that is ultimately a good thing no?

**Comment by d-chambers** [comment]
By the way,  has anyone tested if ObsPy works with freethreaded python? Could be game changer for processing streams with many traces,  but I don't know how much trouble ObsPy's C extensions would cause here.

**Comment by megies** [comment]
OK, so it was a mistake to not cull dependency versions much earlier and it got overlooked also when merging #3333 that it would've been better to kick out the old Python's then and there. It is how it is. While I think it's not the right thing to do to keep support for those old Python versions, since it's last minute we can try to keep them for the release just for the sake of trying to get it done today(ish) finally, while we're at it, without stepping on somebody's toes.

I do think the first thing we should do after is set out a clear policy like NEP29 with exact details what versions will be supported at what point without having to have discussions later on again.

**Comment by ThomasLecocq** [comment]
@filefolder when you'll wake up, might be some earthscope stuff to crosscheck: https://github.com/obspy/obspy/actions/runs/22766605589/job/66035988317#step:10:1899

**Comment by ThomasLecocq** [comment]
**[1.5.0 rc1](https://test-files.pythonhosted.org/packages/81/0b/d0d43f664f2327c5300b9f86302e354141f82701e590597dddbce6a6d903/obspy-1.5.0rc1.tar.gz) building RIGHT NOW**

https://github.com/obspy/obspy/actions/runs/22768000077

Once they will be built, thee wheels & source dist will go https://test.pypi.org/project/obspy/1.5.0rc1/#files

Let's check them out in the coming days - and by mid of next week put a final number on it !

PARTYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYY

**Comment by claudiodsf** [comment]
Congratulations!!!🎊🍾🎈 

Are you sending out an announcement for RC1, to encourage potential testers?

**Comment by megies** [comment]
> Are you sending out an announcement for RC1, to encourage potential testers?

on discourse? we could I guess, if people even read that 😅

**Comment by claudiodsf** [comment]
I can repost on Bluesky 🦋

**Comment by megies** [comment]
[conda-forge](https://github.com/conda-forge/obspy-feedstock/pull/59) happy first try with RC1 😮  (no 3.14 built yet, migration probably triggers after getting the main branch updated with 1.5.0)

**Comment by ThomasLecocq** [comment]
![IMG_20260306_233749.jpg](https://github.com/user-attachments/assets/50e40f23-e229-48d7-bc7a-caad86542fe3)

Arrived in Brussels! Treating myself with a proper ObsPy Release Beer (Westvleteren)
